src/app/containers/Preferences/General/RotaryAxis.js

```diff
@@ -13,7 +15,21 @@ const RotaryAxis = ({ state }) => {
     const { units } = state;
     //TODO
     const handleRotaryAxisToggle = () => {
+        const firmwareType = get(reduxStore.getState(), 'controller.type', 'grbl');
+        const isGrbl = firmwareType.toLocaleLowerCase() === 'grbl';
         const shouldEnableRotary = !rotaryAxisStatus;
+
+        //If grbl, turn off Rotary Axis and update user
+        if (isGrbl) {
+            store.set('rotaryAxisStatus', false);
+            Toaster.clear();
+            Toaster.pop({
+                type: TOASTER_WARNING,
+                msg: 'Your firmware does not support Rotary Axis.',
+            });
+            return;
+        }
+
```

Comment:
The guard doesn't need to happen here, please remove all the firmware check logic here and all. The rest of the guards on the jog controls look good